client/src/Router.tsx

```diff
@@ -1,22 +1,46 @@
-import React from 'react';
-import { Route, Switch, Redirect } from 'react-router-dom';
+import React, { useEffect } from 'react';
+import { Route, Switch, Redirect, useLocation } from 'react-router-dom';
 import LandingPage from './pages/LandingPage';
 import MainPage from './pages/MainPage';
 import WorkPage from './pages/WorkPage';
-import { useRecoilValue } from 'recoil';
-import user from '@/recoil/user';
+import { useRecoilValue, useSetRecoilState } from 'recoil';
+import userAtom from '@/recoil/user';
 import AdminPage from './pages/AdminPage';
 import LogInPage from './pages/LogInPage';
-import { SignUpPage } from './pages/SignUpPage';
+import SignUpPage from './pages/SignUpPage';
+import { getUser } from './lib/api/user';
+import { taskSortByUpdate } from './lib/utils/sort';
+import { UserState } from './contexts/userContext';
+import { useMemo } from 'react';
 
 const Router = () => {
-  const userState = useRecoilValue(user);
+  const userState = useRecoilValue(userAtom);
+  const setUserState = useSetRecoilState(userAtom);
```

Comment:
getter와 setter 모두 쓸 경우에는 
```suggestion
  const [userState, setUserState] = useRecoilState(userAtom);
```
와 같이 리액트처럼 쓰실 수 있습니다.

Reply:
역시 갓 리코일....